internal/cloud/backend_saved_plan_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cloud
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func TestCloud_loadBasic(t *testing.T) {
+
+	bookmark := SavedPlanBookmark{
+		RemotePlanFormat: 1,
+		RunID:            "run-GXfuHMkbyHccAGUg",
+		Hostname:         "app.terraform.io",
+	}
+
+	result, err := LoadSavedPlanBookmark("./testdata/plan-bookmark/bookmark.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if diff := cmp.Diff(bookmark, result, cmp.Comparer(cty.Value.RawEquals)); diff != "" {
+		t.Errorf("wrong result\n%s", diff)
+	}
+}
+
+func TestCloud_saveBasic(t *testing.T) {
+	tmp := t.TempDir()
+	defer os.RemoveAll(tmp)
```

Comment:
Minor issue: This RemoveAll isn't needed — that's the benefit of using `t.TempDir()`, is that it cleans itself up automatically. ([source reference](https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/testing/testing.go;l=1163))

Reply:
Not sure! 🤷 It looks like there's a mix of using it and not using it, in conjunction with t.TempDir(). Probably folks weren't aware of that, or were just copying someone who wasn't aware of that.